umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/instrument.clj

```diff
@@ -82,42 +87,58 @@
 
 (defn xml-elem->instrument
   "Returns instrument record from the instrument element."
-  [doc base-xpath instrument-elem]
-  (let [child-instruments (get-child-instruments doc base-xpath instrument-elem)]
-    (util/remove-nil-keys
-     (if (> (count child-instruments) 0)
-       {:ShortName (value-of instrument-elem iso/short-name-xpath)
-        :LongName (value-of instrument-elem iso/long-name-xpath)
-        :Characteristics (char-and-opsmode/parse-characteristics instrument-elem)
-        :Technique (without-default-value-of instrument-elem "gmi:type/gco:CharacterString")
-        :NumberOfInstruments (count child-instruments)
-        :ComposedOf child-instruments
-        :OperationalModes (char-and-opsmode/parse-operationalmodes instrument-elem)}
-       {:ShortName (value-of instrument-elem iso/short-name-xpath)
-        :LongName (value-of instrument-elem iso/long-name-xpath)
-        :Characteristics (char-and-opsmode/parse-characteristics instrument-elem)
-        :Technique (without-default-value-of instrument-elem "gmi:type/gco:CharacterString")
-        :OperationalModes (char-and-opsmode/parse-operationalmodes instrument-elem)}))))
+  [doc base-xpath alt-xpath? instrument-elem]
```

Comment:
This is not a request for a change, but something to consider:

For refactorings like this where a new parameter is added, it is good practice to do something like the following ...

1. Keep the original arity with the same parameters (this is an easy path for backwards compatibility and helps reduce the number of surprises due to the change to nearly zero).
1. Add a new a new arity N+1 block to the function, where the new parameter is an "options" map containing the key for the new data you want to pass (you can do this in the args with destructuring). Note that this is an excellent means of future-proofing your code.
1. Add the new implementation that a) has the exact same behaviour as the original function when the options map is nil (or the expected options key isn't present, etc.), and b) has the new behaviour when the expected key _is_ present, then
1. Delete the original arity code and in that back simple call the arity N+1 function with the options map being `{}`.

Another nice benefit of this approach is that it makes it super-easy for reviewers to see what has changed, and clearly identify the backwards compatibility plan :-)

Reply:
Btw, step 4 as outlined above is for when the logic is basically identical, modulo small variance due to the options. If the logic is completely different, you just keep the original in the first arity block, and the new logic in the secon arity block.

Reply:
Done.